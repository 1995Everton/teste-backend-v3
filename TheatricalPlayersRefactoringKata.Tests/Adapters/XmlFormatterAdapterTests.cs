using ApprovalTests;
using ApprovalTests.Reporters;
using AutoMapper;
using System.Collections.Generic;
using TheatricalPlayersRefactoringKata.Application.Adapters;
using TheatricalPlayersRefactoringKata.Application.Mappings;
using TheatricalPlayersRefactoringKata.Entities;
using Xunit;

namespace TheatricalPlayersRefactoringKata.Tests.Adapters
{
    public class XmlFormatterAdapterTests
    {
        private readonly IMapper _mapper;

        public XmlFormatterAdapterTests()
        {
            var config = new MapperConfiguration(cfg => {
                cfg.AddProfile<MappingProfile>();
            });

            _mapper = config.CreateMapper();
        }

        [Fact]
        [UseReporter(typeof(DiffReporter))]
        public void Format_ShouldReturnFormattedStatement()
        {
            // Configura um cenário onde uma declaração é formatada em XML.
            // Verifica se a declaração formatada está correta.

            // Arrange
            var formatter = new XmlFormatterAdapter(_mapper);
            var statement = new Statement
            {
                Customer = "John Doe",
                Items = new List<StatementItem>
                    {
                        new StatementItem { Name = "Hamlet", AmountOwed = 400.00m, Seats = 55 },
                        new StatementItem { Name = "As You Like It", AmountOwed = 300.00m, Seats = 35 },
                        new StatementItem { Name = "Othello", AmountOwed = 500.00m, Seats = 40 }
                    },
                TotalAmountOwed = 1200.00m,
                TotalEarnedCredits = 30
            };

            // Act
            var result = formatter.Format(statement);

            // Assert
            Approvals.Verify(result);
        }
    }
}

