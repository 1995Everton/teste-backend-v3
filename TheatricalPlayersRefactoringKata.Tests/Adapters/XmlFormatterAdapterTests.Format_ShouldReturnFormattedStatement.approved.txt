<?xml version="1.0" encoding="utf-8"?>
<Statement xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xmlns:xsd="http://www.w3.org/2001/XMLSchema">
  <Customer>John Doe</Customer>
  <Items>
    <Item>
      <AmountOwed>400.00</AmountOwed>
      <EarnedCredits>0</EarnedCredits>
      <Seats>55</Seats>
    </Item>
    <Item>
      <AmountOwed>300.00</AmountOwed>
      <EarnedCredits>0</EarnedCredits>
      <Seats>35</Seats>
    </Item>
    <Item>
      <AmountOwed>500.00</AmountOwed>
      <EarnedCredits>0</EarnedCredits>
      <Seats>40</Seats>
    </Item>
  </Items>
  <AmountOwed>1200.00</AmountOwed>
  <EarnedCredits>30</EarnedCredits>
</Statement>